sdk/transaction-error/src/lib.rs

```diff
@@ -137,6 +137,9 @@ pub enum TransactionError {
 
     /// Program cache hit max limit.
     ProgramCacheHitMaxLimit,
+
+    /// Commit failed internally.
+    CommitFailed,
```

Comment:
Why are we now able to fail on commit while it was not previously possible?

Reply:
It seems like we are failing the pre-commit check (recording), not commit itself.

Reply:
If that's the case, why can we not eat the error in the same way we do in normal block-production, without adding a new variant here?